packages/http/src/Adaptor.js

```diff
@@ -43,87 +35,69 @@ export function execute(...operations) {
   };
 }
 
-var cookiejar = new tough.CookieJar();
-var Cookie = tough.Cookie;
+function handleResponse(state, response) {
+  return {
+    ...composeNextState(state, response.body),
+    response,
+  };
+}
 
-axios.interceptors.request.use(config => {
-  cookiejar?.getCookies(config.url, (err, cookies) => {
-    config.headers.cookie = cookies?.join('; ');
-  });
-  return config;
-});
+function handleCallback(state, callback) {
+  if (callback) return callback(state);
+  return state;
+}
 
-function handleCookies(response) {
-  const { config, data, headers } = response;
-  if (config.keepCookie) {
-    let cookies;
-    let keepCookies = [];
+function request(method, path, params, callback) {
+  return state => {
+    const [resolvedPath, resolvedParams] = expandReferences(
+      state,
+      path,
+      params
+    );
 
-    if (headers['set-cookie']) {
-      if (headers['set-cookie'] instanceof Array) {
-        cookies = headers['set-cookie']?.map(Cookie.parse);
-      } else {
-        cookies = [Cookie.parse(headers['set-cookie'])];
-      }
+    let body = resolvedParams?.body;
+    let headers = resolvedParams?.headers;
 
-      headers['set-cookie']?.forEach(c => {
-        cookiejar.setCookie(Cookie.parse(c), config.url, (err, cookie) => {
-          keepCookies?.push(cookie?.cookieString());
-        });
+    if (resolvedParams?.form) {
+      let form = new FormData();
+      Object.entries(resolvedParams?.form).forEach(element => {
```

Comment:
I think we can simplify this loop too:
```
 let form = new FormData();
 for (const key of resolvedParams.form) {
     form.append(key, form[key])
 }
```
You shouldn't really need the optional chaining `?` operator on `resolvedParams`, certainly not inside the if block. Consider defaulting `resolvedParams` to {}